kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/SpecManagementServiceImpl.java

```diff
@@ -58,6 +58,10 @@ public synchronized void saveContainerSpec(String serverTemplateId,
             throw new KieServerControllerIllegalArgumentException("No server template found for id " + serverTemplateId);
         }
 
+        if (containerSpec.getId().isEmpty() || containerSpec.getId()==null ) {
```

Comment:
In this case if getId() is null it will throw a NPE, you should check if `containerSpec.getId()==null` before checking `getId().isEmpty()`. Same in code below. To check what I'm saying add a small test just for null values and you will see that behavior

Reply:
Yes, agreed null check should come first.